python_modules/dagster/dagster/_grpc/server.py

```diff
@@ -454,6 +468,8 @@ def ReloadCode(
 
     def Ping(self, request, _context: grpc.ServicerContext) -> api_pb2.PingReply:
         echo = request.echo
+        if self._enable_metrics:
+            _record_utilization_metrics(self._logger)
```

Comment:
Can we centralize this in the retrieve_metrics decorator? Would prefer that decorator to be the source of truth for how `_UTILIZATION_METRICS` is mutated.